Components/Queues/Handler/PhpAmqpHandler.php

```diff
@@ -0,0 +1,232 @@
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\QueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageHandlerInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\IsStopableConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointFactory;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+
+/**
+ * Class PhpAmqpHandler
+ * @package Smartbox\Integration\FrameworkBundle\Components\Queues\Handler
+ */
+class PhpAmqpHandler implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+    use UsesSmartesbHelper;
+    use UsesExceptionHandlerTrait;
+    use IsStopableConsumer;
+
+    /**
+     * @var EndpointInterface
+     */
+    private $endpoint;
+
+    /**
+     * Format used by serializer function
+     * @var string
+     */
+    private $format;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * @param EndpointInterface        $endpoint   The endpoint used to consume
+     * @param int                      $expirationCount  Maximum amount of message to consume before stopping
+     * @param string                   $format     json|xml|php
+     * @param null $serializer Serializer to use for deserialization
+     */
+    public function __construct(EndpointInterface $endpoint, int $expirationCount = -1, string $format = QueueDriverInterface::FORMAT_JSON, $serializer = null)
+    {
+        $this->endpoint = $endpoint;
+        $this->setExpirationCount($expirationCount);
+        $this->format = $format;
+        $this->serializer = $serializer ?? SerializerBuilder::create()->build();
+    }
+
+    /**
+     * Consumes the message and dispatch it to others features
+     * @param string $consumerTag
+     * @param AMQPChannel $channel
+     * @param string $queueName
+     * @throws \Exception
+     */
+    public function consume(string $consumerTag, AMQPChannel $channel, string $queueName)
+    {
+        $callback = function($message) use ($channel) {
+            $this->isConnected($channel);
+            // Send a message with the string "quit" to cancel the consumer.
+            if ($message->body === 'quit' || $this->shouldStop()) {
+                $channel->basic_cancel($message->delivery_info['consumer_tag']);
+                return false;
```

Comment:
Any reason to return `false` here? Nothing seems to be using this boolean.

Reply:
Yes, to inform that the consume fails and stop the execution.